boa_engine/src/builtins/date/mod.rs

```diff
@@ -547,8 +540,19 @@ impl Date {
     ///
     /// [spec]: https://tc39.es/ecma262/#sec-date.prototype.getdate
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate
-    pub fn get_date(&self) -> f64 {
-        self.to_local().map_or(f64::NAN, |dt| f64::from(dt.day()))
+    pub fn get_date(this: &JsValue, _args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        if let Some(t) = this_time_value(this, context)?.0 {
+            let date = Local::now()
+                .timezone()
+                .from_utc_datetime(&t)
+                .format("%Y-%m-%dT%H:%M:%S%.f%:z")
+                .to_string();
+            let local = NaiveDateTime::parse_from_str(&date, "%Y-%m-%dT%H:%M:%S%.f%:z")
+                .expect("INFALLIBLE: Converting string built from `Local::now()` to NaiveDateTime");
```

Comment:
Why do we have to do the conversion from a `DateTime<Local>` to a `String` and then to a `NaiveDateTime`?
Should this not yield the same result:
```rust
Local::now()
    .timezone()
    .from_utc_datetime(&t)
    .day()
```

At least on my setup the `test/built-ins/Date/prototype/getDate/` testsuite passes and I'm not on UTC.

Same for all the other non-utc get methods.

Reply:
I should have read `chrono` docs properly, as I was trying to call `day()` after `format()`. So I used a workaround.